server/src/db/schemas/favorites.schema.ts

```diff
@@ -0,0 +1,24 @@
+import mongoose from "mongoose";
+import { UserModel } from "./user.schema";
+import { ItemModel } from "./item.schema";
+
+const Schema = mongoose.Schema;
+
+export interface Favorite {
+    id?: mongoose.Types.ObjectId
+    userId: mongoose.Types.ObjectId,
+    itemId: mongoose.Types.ObjectId,
+}
+
+const FavoriteModelSchema = new Schema<Favorite>({
+    userId: {
```

Comment:
If we compare this to `wishlist` in `Series Social Network`, we saved the record as `{ _id, userId, items }`.
The reason behind this was to send the full wishlist/favorites to the user after each add/remove operation. That way the favorites is "in sync" between different devices who make add/remove operations across the application.

I suggest updating the schema that way. Look at `Series Social Network`, you've already done the whole work in the past.